posit-dev/positron
# Add error relaying on new folder modal dialog for errors that occur on folder creation attempt.

Addresses  #3424
<img width="432" alt="image" src="https://github.com/posit-dev/positron/assets/6764693/44bcddfa-1351-4b6b-af73-e462a8de7baf">

Related to #3422 but with a scope of general errors in the new folder creation. 

### Intent

Relay error messages to the user when a new folder creation attempt fails. 

### Approach

There are two approaches to solving this included in this PR as the two commits. 

[a2fe50d](https://github.com/posit-dev/positron/pull/3423/commits/a2fe50dae90904c56fafcb29ce2fbca714fcf07d): 
The first is a more hands-on approach where we manually wrap the on accept behavior in a try-catch statement and have an error message in a `useState()` that forwards the message. The logic is repeated across the new folder and new folder from git modals. 

bbf596768b54bd8a263a93c4b5e8768ba6edea15: The second moves this try-catch logic down a level into the `OKCancelModalDialog` component itself by adding a `catchErrors` prop that does the same thing for us. The end behavior of both is the same. 


Wrap the onAccept logic in a try-catch and forward the error to the modal instead of just closing without doing anything. 

### QA Notes

I added a css property to the modal dialog content area to allow for scrolling when we have long error messages. There's a chance this causes issues with other modals with unexpected scroll bars etc. That being said, I didnt notice anything.
